src/python/essentia/pytools/batch_extractors.py

```
@@ -0,0 +1,219 @@
+                      generate_log=True, audio_types=None, skip_analyzed=False,
+                      jobs=0, verbose=True):
+    if not audio_types:
+        audio_types = ('.wav', '.aiff', '.flac', '.mp3', '.ogg')
+        print("Audio files extensions considered by default: " +
+              ' '.join(audio_types))
+    else:
+        if type(audio_types) == str:
+            audio_types = [audio_types]
+
+        audio_types = tuple(audio_types)
+        audio_types = tuple([i.lower() for i in audio_types])
+        print("Searching for audio files extensions: " + ' '.join(audio_types))
+    print("")
+
+    output_dir = os.path.abspath(output_dir)
+    audio_dir = os.path.abspath(audio_dir)
+
+    if jobs == 0:
+        try:
+            jobs = cpu_count()
+        except NotImplementedError:
+            print('Failed to automatically detect the cpu count, '
+                  'the analysis will try to continue with 4 jobs. '
+                  'For a different behavior change the `job` parameter.')
+            jobs = 4
+
+    skipped_count = 0
+    skipped_files = []
+    cmd_lines = []
+    for root, _, filenames in os.walk(audio_dir):
+        for filename in filenames:
+            if filename.lower().endswith(audio_types):
+                audio_file = os.path.join(audio_dir, root, filename)
+                out_file = os.path.join(output_dir, output_dir, filename)
+
+                if skip_analyzed:
+                    if os.path.isfile( '{}.{}'.format(out_file, output_extension)):
+                        print("Found descriptor file for " +
+                              audio_file + ", skipping...")
+                        skipped_files.append(audio_file)
+                        skipped_count += 1
+                        continue
+                folder = os.path.dirname(out_file)
+                if not os.path.exists(folder):
+                    os.makedirs(folder)
+
+                elif os.path.isfile(folder):
```

Is this a typically error that people might get? I would just typically do a check if it exists, regardless of if it's a file or a directory, so just a simple `os.path.exists`, and raise an error if necessary?

---

We just want to raise an error if it is a file. For the folder case, it can be the result of a previous incomplete analysis that the user wants to resume.